logictest: add cockroach-go-testserver-configs

RaduBerinde opened the pull request:
This change adds a `cockroach-go-testserver-configs` which includes
all the go testserver configs (one for each supported older version).

This allows us to move versions forward without having to modify the
`LogicTest:` directive in each test.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136902)
<!-- Reviewable:end -->

RaduBerinde committed:
logictest: add cockroach-go-testserver-configs

This change adds a `cockroach-go-testserver-configs` which includes
all the go testserver configs (one for each supported older version).

This allows us to move versions forward without having to modify the
`LogicTest:` directive in each test.

Epic: none
Release note: None

rafiss commented on pkg/sql/logictest/testdata/logic_test/upgrade_skip_version:
```diff
@@ -1,19 +0,0 @@
-# TODO(radu): reenable this test in 25.2 when we support upgrade from two
-# previous releases (#134455).
-## LogicTest: cockroach-go-testserver-24.2
```
shouldn't we be testing an upgrade from 24.2 directly to 25.1 now?

RaduBerinde replied:
No, we won't support that - the minimum supported version is 24.3. We will allow 24.3 to 25.2.

rafiss replied:
it would be nice to keep this file but just have it skipped or disabled somehow so we are reminded to re-add this test.

RaduBerinde replied:
I did keep it, I just renamed it to "upgrade" since it now checks upgrade from all supported versions. So it will automatically run skip upgrades when we support them.
<!-- Reviewable comment -ODqKrcZ7d3RCqkD89nH:b-4xvbww -->
<!-- Sent from Reviewable.io -->

rafiss replied:
ah thanks. github collapsed the file and i would have expected it to identify this as a rename.

RaduBerinde reviewed (commented):
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
